### ISSUE TITLE
🆕 Software Suggestion | User-Agent Switcher and Manager firefox addon

### ISSUE THREAD

**Issue by sell** [privacytoolsIO/privacytools.io#1668]
## Basic Information

**Name:** User-Agent Switcher and Manager
**Category:** Browser Addons
**URL:** https://addons.mozilla.org/en-US/firefox/addon/user-agent-string-switcher/
**Source:** https://github.com/ray-lothian/UserAgent-Switcher/ 

## Description
Allows you to spoof your user agent.

## Why I am making the suggestion
This suggestions, helps with sites not be able to track what kind of browser or pc you are on. For instance, you can me on mac and spoof your user-agent so it thinks you are on windows using firefox. 

## My connection with the software
Not affiliated at all with this software/addon.

- [x] I will keep the issue up-to-date if something I have said changes or I remember a connection with the software.

**Comment by Thorin-Oakenpants** [comment]
This won't stop JS from detecting that you're using Firefox. It won't fully hide your major OS, and in fact on many Linux distros, it won't hide your distro. It won't even fully hide your version. It can't even totally hide the Firefox release channel. No extension can. Not even Tor Browser can. @Mikaela knows what I'm talking about, right MIkaela (ssssh, don't tell)?

Maybe it can defeat some rudimentary scripts - but scripts that aren't so simple - will just unmask you as super unique.

There's also lots of `information paradoxes` that can happen when you spoof - and when these happen you just become unique. For example, until a week ago, Chameleon hadn't been spoofing at least one navigator property it should have been when Firefox users were trying to be non-Firefox.

And in a fingerprinting script that has a lot of things (such as amiunique), if you're trying to not be Firefox, then all the other factors combined make you unique - i.e why does your browser not have x feature when you say you're chrome, or it has a feature that only Firefox has. There are literally thousands of these, and amiunique will some some of these since it's querying/built-for multiple browsers - i.e it is not what I would call a `targeted` FP script - just a general one.

Additionally, there are a lot of holes to cover. CanvasBlocker until a month ago, leaked it's UA spoof via window.open (which I pointed out, and it got fixed) - I'm guessing most other UA spoofing extensions don't have this covered. Other extensions in the past screwed up covering basics like iframes, and other methods (there's lots of them).

tl;dr: you are better off using `privacy.resistFingerprinting`, or if you don't like that, then just being on Firefox stable release is your best move. Anything else and you **will** become unique (I'm assuming you would spoof as non-Firefox, otherwise why spoof?)

---

Whatever PTIO decides, please do NOT promote this an anti-FPing extension

I'm not going to test or install the extension in question, so I don't know if it uses whitelists (the screenshots don't seem to show any). i.e. it can sometimes be handy to only apply a spoof to a particular site - default do nothing, only spoof when you add a rule. If it did this, and only this, then it could be useful for the following reasons
- for compat reasons: did you know Mozilla have a compat team? And you can tell them, and they can get it fixed for you? As a last resort they can add a UA override to the webcompat system addon
   - type `about:compat` into your urlbar
   - [moz wiki on compat](https://wiki.mozilla.org/Compatibility)
   - [feb 2019 ghacks article on about:compat](https://www.ghacks.net/2019/02/28/firefoxs-new-web-compatibility-page/)
- handy for website dev work / testing I guess: but IMO sites should be using feature detection, not UA sniffing

I also don't think that compat and website dev is PTIO's aim, so I personally, I wouldn't include it.

**Comment by sell** [comment]
Thanks for the very detailed response, and I spoof the system and the browser with the extension. It does work, with sites as they do think I am on a different browser and different operation system, but as you stated above it make me super unique. I can tell you are very smarting than I am and I will just have to take your word for it. 

It has been listed on many sites about it being a essential, also there has been nothing but good about the addon, here is one review from [alternativeto.net](https://alternativeto.net/list/381/top-security-web-browser-plugins)

Once again, thanks for that response @Thorin-Oakenpants 👍

**Comment by Mikaela** [comment]
Thank you again @Thorin-Oakenpants, I think we can close this.

> It won't even fully hide your version. It can't even totally hide the Firefox release channel. No extension can. Not even Tor Browser can. @Mikaela knows what I'm talking about, right MIkaela (ssssh, don't tell)?

In the intersts of transparency I have to say that I believe you are talking about [TorZillaPrint](https://ghacksuserjs.github.io/TorZillaPrint/TorZillaPrint.html) which with I understand you to be affiliated with.

**Comment by Thorin-Oakenpants** [comment]
@Mikaela : affiliated with? - more like 95% me. It's my own little project. And yes, I just don't want to necessarily blurt out it's existence to everyone because most people won't understand the math and mechanics behind a lot of it and make false assumptions / accusations - tl;dr: I can't be assed with dealing with people, so I'd rather it wasn't blurted out :)

Feel free to edit your post and delete the edit history - up to you, but I did say `ssssh, don't tell` :)
